src/test/java/com/liferay/faces/util/cache/CacheTest.java

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright (c) 2000-2017 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.util.cache;
+
+import java.util.Queue;
+import java.util.Random;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.liferay.faces.util.cache.internal.CacheFactoryImpl;
+
+
+/**
+ * @author  Kyle Stiemann
+ */
+public class CacheTest {
```

Comment:
@stiemannkj1 If the JavaDoc is modified so that `throws IllegalArgumentException` is documented for factory methods that take an `initialCapactity` or `maxCapacity` then there should be some assertions that verify that `IllegalArgumentException` is actually thrown.